chore: support legacy BLS CLI

spypsy opened the pull request:
When using the `valkeys` CLI, users expected `--account-index` to affect the resulting BLS keys. However this was only used for ETH key derivation (while `--bls-path` was expected for BLS) which mean some users ended up creating duplicate BLS keys and running into issues when staking. 
However, in order to support `--count` for BLS keys, we did use `addressIndex` which is what was appended when using `--count`. This was used in the derivation path like: `m/12381/3600/<address_index>/0/0`

To fix this, in #18430, we started using `--account-index` & `--address-index` for BLS path derivation and used them in the path like: `m/12381/3600/<account_index>/0/<address_index>`.

This meant that users with either used `--count` or `--address|account-index` in 2.1.4, are getting a different result in 2.1.5 onwards.

Because some may need to re-create their keys, in order to make it easier we have added env var `LEGACY_BLS_CLI` which, if set to `1` \ `true`, will run the code path that was in 2.1.4, thus generating the same result.

Additional QoL changes:
- rename `--out-dir` to `--encrypted-keystore-dir` for clarity
- `aztec staker` was outputting a JSON file with an array of attesters while `aztec valkeys --staker-output` was outputting multiple JSON files. Update so both output multiple

Also, some QoL changes:
- change `out-dir` to `encrypted-keystore-dir` for clarification
- use correct output dir for staker output (was previously using out-dir isntead of data-dir)
- consistent output between `aztec staker` & `aztec valkeys new --staker-output`. We were previously outputting a single JSON file with an array in `aztec staker` and multiple JSON files with `--staker-output`

alexghr commented on yarn-project/cli/src/cmds/validator_keys/add.ts:
```diff
@@ -36,7 +40,7 @@ export async function addValidatorKeys(existing: string, options: AddValidatorKe
     remoteSigner: remoteSignerOpt,
     password,
-    outDir,
+    encryptedKeystoreDir,
```
oh was `--out-dir` only used for storing encrypted keystores? I see maybe not so important to have an alias

spypsy committed:
chore: support legacy BLS CLI

When using the `valkeys` CLI, users expected `--account-index` to affect the resulting BLS keys. However this was only used for ETH key derivation (while `--bls-path` was expected for BLS) which mean some users ended up creating duplicate BLS keys and running into issues when staking.
However, in order to support `--count` for BLS keys, we did use `addressIndex` which is what was appended when using `--count`. This was used in the derivation path like: `m/12381/3600/<address_index>/0/0`

To fix this, in #18430, we started using `--account-index` & `--address-index` for BLS path derivation and used them in the path like: `m/12381/3600/<account_index>/0/<address_index>`.

This meant that users with either used `--count` or `--address|account-index` in 2.1.4, are getting a different result in 2.1.5 onwards.

Because some may need to re-create their keys, in order to make it easier we have added env var `LEGACY_BLS_CLI` which, if set to `1` \ `true`, will run the code path that was in 2.1.4, thus generating the same result.

Additional QoL changes:
- rename `--out-dir` to `--encrypted-keystore-dir` for clarity
- `aztec staker` was outputting a JSON file with an array of attesters while `aztec valkeys --staker-output` was outputting multiple JSON files. Update so both output multiple

Also, some QoL changes:
- change `out-dir` to `encrypted-keystore-dir` for clarification
- use correct output dir for staker output (was previously using out-dir isntead of data-dir)
- consistent output between `aztec staker` & `aztec valkeys new --staker-output`. We were previously outputting a single JSON file with an array in `aztec staker` and multiple JSON files with `--staker-output`
